common/shareddb.cpp

```diff
@@ -1448,6 +1448,105 @@ bool SharedDatabase::LoadNPCFactionLists(const std::string &prefix) {
 	return true;
 }
 
+void SharedDatabase::GetSplashFactionInfo(uint32 &list_count, uint32 &max_lists)
+{
+	list_count = 0;
+	max_lists = 0;
+
+	const std::string query = "SELECT COUNT(*), MAX(id) FROM splash_faction";
+	auto results = QueryDatabase(query);
+	if (!results.Success()) {
+		return;
+	}
+
+	if (results.RowCount() == 0)
+		return;
+
+	auto row = results.begin();
+
+	list_count = static_cast<uint32>(atoul(row[0]));
+	max_lists = static_cast<uint32>(atoul(row[1] ? row[1] : "0"));
+}
+
+const SplashFactions *SharedDatabase::GetSplashFactionHit(int id)
+{
+	if (!splash_faction_hash) {
+		return nullptr;
+	}
+
+	if (splash_faction_hash->exists(id)) {
+		return &(splash_faction_hash->at(id));
+	}
+
+	return nullptr;
+}
+
+void SharedDatabase::LoadSplashFaction(void *data, uint32 size, uint32 list_count, uint32 max_lists)
+{
+	EQ::FixedMemoryHashSet<SplashFactions> hash(reinterpret_cast<uint8 *>(data), size, list_count, max_lists);
+	SplashFactions faction;
+
+	// yes, this table is kind of shit, but it works and given how VI wrote somethings *cough* spells *cough* it's
+	// probably what they do :P
+	const std::string query =
+	    "SELECT id, splash_id1, splash_mod1, splash_id2, splash_mod2, splash_id3, splash_mod3, splash_id4, "
+	    "splash_mod4, splash_id5, splash_mod5, splash_id6, splash_mod6, splash_id7, splash_mod7, splash_id8, "
+	    "splash_mod8, splash_id9, splash_mod9, splash_id10, splash_mod10 FROM splash_faction";
```

Comment:
Should be able to just do `auto results = SplashFactionRepository::All(*this)` and loop the returned vector instead of manually constructing the query

Reply:
Agreed